lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -138,6 +146,15 @@
         GetObject permission but no other permissions. In this case using the option mode: get will fail without specifying
         ignore_nonexistent_bucket: True."
     version_added: "2.3"
+  use_v4_signature:
```

Comment:
I'm not sure about this because the HeadObject permission isn't required for `mode: get`, just `mode: getstr`

Reply:
I added 275cab51bf1468c4c86432eb2627eb32bff1f23e so the user doesn't need to declare this for put operations but I'm not I'm happy with it. Maybe instead C(use_v4_signature) should be required together with encryption_kms_key_id and otherwise the user is expected to know when they need it? It would be nicer for the module to handle this, I'm just not sure how since head_object() is not guaranteed. @ryansb What do you think?